avalanche/evaluation/metric_definitions.py

```diff
@@ -212,11 +210,11 @@ def __init__(self, metric, reset_at='experience', emit_at='experience',
         super(GenericPluginMetric, self).__init__()
         assert mode in {'train', 'eval'}
         if mode == 'train':
-            assert reset_at in {'iteration', 'epoch'}
```

Comment:
The idea behind the asserts was that if the metric works at training time it should reset and emit values only during training stages like `iteration` and `epoch`. Correspondingly, if the metric works at eval time it should reset and emit at `experience` or `stream`. Do you need to emit or reset in different stages?

Reply:
Yes for eg the LabelRepartition needs to be emited at stream / experience, it would be unnecessary to emit it every epoch, as we would emit the same thing each time, it only depends on the dataset

The methods within the class already allow that
For eg the implementation of `GenericPluginMetric .after_training` expects `self._emit_at == 'stream' and self._mode == 'train'`

Reply:
Since you need to emit at `experience` or `stream` wouldn't be better to set this as a `eval` metric? In this way we can maintain these lines as before. This would also bypass the training transform problem I think, since you would not operate during training mode.

Reply:
Does that mean only getting the info from the eval stream?
The eval and train label repartitions won't be the same, and having the train imbalance (due to limited/no memory) is something we want to be able to plot

Reply:
You are right, I think we can proceed like you did, thanks.